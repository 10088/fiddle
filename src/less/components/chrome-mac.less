.darwin {
  .chrome {
    text-align: center;
    background-color: @background-1;
    border-bottom: @border;
    -webkit-app-region: drag;
    height: 23px;
    font-size: 1.4rem;

    small,
    p,
    span {
      user-select: none;
    }
  }
}
