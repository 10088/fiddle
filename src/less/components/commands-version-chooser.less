button.version-chooser.bp3-button {
  border-radius: 3px;
}
