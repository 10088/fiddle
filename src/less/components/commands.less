@import "../variables.less";

header {
  box-shadow: 0 0 0 1px rgba(16, 22, 26, 0.2), 0 0 0 rgba(16, 22, 26, 0), 0 1px 1px rgba(16, 22, 26, 0.4);
  background-color: @background-1;
  -webkit-app-region: drag;
}

.commands {
  display: flex;
  justify-content: space-between;
  height: 50px;
  padding: 10px;
  box-sizing: border-box;
  font-family: @fonts-common;

  button,
  input,
  select {
    -webkit-app-region: no-drag;
    margin-bottom: 0;
  }

  select {
    width: 100px;
    margin-left: 5px;
  }

  & > div {
    display: flex;

    &:last-of-type {
      margin-right: 10px;

      button {
        margin-left: 5px;
      }
    }
  }

  .address-bar {
    height: 28px;
    width: 360px;
    transition: width 0.4s;
    display: flex;

    input,
    span {
      font-family: @fonts-common;
      font-weight: 300;
      font-size: 11px;
      cursor: auto;
      -webkit-app-region: no-drag;
    }

    input {
      flex-grow: 1;
      background: white;
      padding-left: 1rem;
      height: 2.6rem;
    }

    input:invalid {
      box-shadow: inset 0 0 0 1px #e4252d;
    }

    span {
      background: white;
      color: #1d1d1d;
      border-bottom-color: rgb(209, 209, 209);
      border-bottom-style: solid;
      border-bottom-left-radius: 4px;
      border-bottom-width: 1px;
      border-image-outset: 0;
      border-image-repeat: stretch;
      border-image-slice: 100%;
      border-image-source: none;
      border-image-width: 1;
      border-left-color: rgb(209, 209, 209);
      border-left-style: solid;
      border-left-width: 1px;
      border-top-color: rgb(209, 209, 209);
      border-top-left-radius: 4px;
      border-top-style: solid;
      border-top-width: 1px;
      display: inline-block;
      height: 1.2rem;
      letter-spacing: normal;
      margin: 0;
      padding: 6px 0 10px 10px;
      user-select: text;
      width: 132px;
      text-rendering: auto;
      text-shadow: none;
      text-transform: none;
    }

    &.empty {
      width: 200px;
    }

    &:focus-within {
      span,
      input {
        border-color: @foreground-2;
      }
    }
  }
}
