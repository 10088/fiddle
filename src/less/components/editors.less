.mosaic-tile {
  animation: fadein 0.15s;
}

.mosaic-window-toolbar  {
  & > div {
    display: flex;
    justify-content: space-between;
    width: 100%;
    padding: 0 3px 0 29px;

    .mosaic-controls {
      transform: scale(0.7);
    }

    h5 {
      margin: 4px 0 0 0;
    }

    button {
      margin-left: 8px;
    }
  }
}

@keyframes fadein {
  from { opacity: 0; }
  to { opacity: 1; }
}
