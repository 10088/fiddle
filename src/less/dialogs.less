@import "variables.less";

.dialogs {
  z-index: 10;
  position: fixed;
  height: 100vh;
  width: 100vw;
  pointer-events: none;

  * {
    pointer-events: all;
  }

  .generateTokenText svg {
    margin-right: 4px;
  }
}

.darwin {
  .dialogs {
    padding-top: 23px;
  }
}

.tokenDialog {
  position: absolute;
  top: 60px;
  width: 300px;
  padding: 10px;
  z-index: 100;
  background-color: @electron-background-3;
  border: @default-border;
  color: #f3f3f3;
  border-radius: 6px;
  left: -10000px;
  opacity: 0;
  transition: top 0.4s ease-in-out, opacity 0.4s ease-in-out;

  &.tokenDialogVisible {
    left: calc(50% - (300px / 2));
    opacity: 1;
    top: 100px;
  }

  button {
    color: #1e2527;
    float: right;
    margin-left: 5px;
    margin-bottom: 0;
  }

  a {
    cursor: pointer;
  }

  input {
    width: 100%;
    color: rgba(159, 234, 249, 0.5);
    height: 28px;
    margin-top: 10px;
    background-color: @electron-background;
    border: 1px solid @electron-dark;
    outline: 0 solid transparent;

    &:focus {
      border-color: @electron-bright;
    }

    &.hasError {
      border-color: #df3434;
    }
  }

  .tokenSpinner {
    position: absolute;
    z-index: 3;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.8);
    display: flex;
    align-items: center;
    justify-content: center;
  }
}

.dialogDrop {
  opacity: 0;
  pointer-events: none;
  position: fixed;
  z-index: 100;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: black;
  transition: opacity 0.4s ease-in-out;

  &.dialogDropVisible {
    pointer-events: all;
    opacity: 0.5;
  }
}
