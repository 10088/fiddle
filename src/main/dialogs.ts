import { dialog } from 'electron';
import { ipcMainManager } from './ipc';
import { IpcEvents } from '../ipc-events';
import { getOrCreateMainWindow } from './windows';

/**
 * Listens to IPC events related to dialogs and message boxes
 *
 * @export
 */
export function setupDialogs() {
  ipcMainManager.on(IpcEvents.SHOW_WARNING_DIALOG, (_event, args) => {
    showWarningDialog(args);
  });
}

/**
 * Shows a warning dialog
 *
 * @param {Electron.MessageBoxOptions} args
 */
function showWarningDialog(args: Electron.MessageBoxOptions) {
  dialog.showMessageBox(getOrCreateMainWindow(), {
    type: 'warning',
    ...args
  });
}
