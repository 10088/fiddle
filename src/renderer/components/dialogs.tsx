import { observer } from 'mobx-react';
import * as React from 'react';

import { AppState } from '../state';
import { AddVersionDialog } from './dialog-add-version';
import { TokenDialog } from './dialog-token';
import { WarningDialog } from './dialog-warning';
import { Settings } from './settings';
import { ConfirmDialog } from './dialog-confirm';

export interface DialogsProps {
  appState: AppState;
}

/**
 * Dialogs (like the GitHub PAT input).
 *
 * @class Dialogs
 * @extends {React.Component<DialogsProps, {}>}
 */
@observer
export class Dialogs extends React.Component<DialogsProps, {}> {
  public render() {
    const { appState } = this.props;
    const { isTokenDialogShowing, isSettingsShowing, isAddVersionDialogShowing } = appState;
    const maybeToken = isTokenDialogShowing
      ? <TokenDialog key='dialogs' appState={appState} />
      : null;
    const maybeSettings = isSettingsShowing
      ? <Settings key='settings' appState={appState} />
      : null;
    const maybeAddLocalVersion = isAddVersionDialogShowing
      ? <AddVersionDialog key='add-version-dialog' appState={appState} />
      : null;
    const eitherWarningOrPrompt = appState.isWarningDialogShowing ? <WarningDialog appState={appState} /> ? appState.isConfirmationPromptShowing ? <ConfirmDialog appState={appState}/> : null;

    return (
      <div key='dialogs' className='dialogs'>
        {maybeToken}
        {maybeSettings}
        {maybeAddLocalVersion}
        {eitherWarningOrPrompt}
      </div>
    );
  }
}
