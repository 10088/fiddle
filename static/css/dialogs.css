.dialogs {
  z-index: 10;
  position: fixed;
  height: 100vh;
  width: 100vw;
  pointer-events: none;
}

.dialogs * {
  pointer-events: all;
}

.tokenDialog {
  position: absolute;
  z-index: 2;
  top: 60px;
  width: 300px;
  padding: 10px;
  z-index: 1000;
  background-color: var(--electron-background-2);
  border: 1px solid var(--electron-dark);
  color: #f3f3f3;
  border-radius: 6px;
  left: -10000px;
  opacity: 0;
  transition: top 0.4s ease-in-out, opacity 0.4s ease-in-out;
}

.tokenDialog.tokenDialogVisible {
  left: calc(50% - (300px / 2));
  opacity: 1;
  top: 100px;
}

.generateTokenText svg {
  margin-right: 4px;
}

.tokenDialog .tokenSpinner {
  position: absolute;
  z-index: 3;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.8);
  display: flex;
  align-items: center;
  justify-content: center;
}

.tokenDialog button {
  color: #1e2527;
  float: right;
  margin-left: 5px;
  margin-bottom: 0;
}

.tokenDialog a {
  cursor: pointer;
}

.tokenDialog input {
  width: 100%;
  color: rgba(159, 234, 249, 0.5);
  height: 28px;
  margin-top: 10px;
  background-color: var(--electron-background);
  border: 1px solid var(--electron-dark);
  outline: 0 solid transparent;
}

.tokenDialog input:focus {
  border-color: var(--electron-bright);
}

.tokenDialog input.hasError {
  border-color: #df3434;
}

.dialogDrop {
  opacity: 0;
  pointer-events: none;
  position: fixed;
  z-index: 1;
  top: 0;
  left: 0;
  width: 100vw;
  height: 100vh;
  background: black;
  transition: opacity 0.4s ease-in-out;
}

.dialogDrop.dialogDropVisible {
  pointer-events: all;
  opacity: 0.5;
}