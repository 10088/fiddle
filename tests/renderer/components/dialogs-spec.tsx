import { shallow } from 'enzyme';
import * as React from 'react';

import { Dialogs } from '../../../src/renderer/components/dialogs';
import { overridePlatform, resetPlatform } from '../../utils';

describe('Dialogs component', () => {
  beforeAll(() => {
    // We render the buttons different depending on the
    // platform, so let' have a uniform platform for unit tests
    overridePlatform('darwin');
  });

  beforeEach(() => {
    this.store = {
      isTokenDialogShowing: false,
      isSettingsShowing: false
    };
  });

  afterAll(() => {
    resetPlatform();
  });

  it('renders initially without visible dialogs', () => {
    const wrapper = shallow(<Dialogs appState={this.store} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.dialogs').html()).toBe('<div class="dialogs"></div>');
  });

  it('renders the token dialog', () => {
    this.store.isTokenDialogShowing = true;
    const wrapper = shallow(<Dialogs appState={this.store} />);
    expect(wrapper.text()).toBe('<TokenDialog />');
  });

  it('renders the settings dialog', () => {
    this.store.isSettingsShowing = true;
    const wrapper = shallow(<Dialogs appState={this.store} />);
    expect(wrapper.text()).toBe('<Settings />');
  });
});
