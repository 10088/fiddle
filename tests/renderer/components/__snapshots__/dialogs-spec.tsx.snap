// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialogs component renders initially without visible dialogs 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Dialogs
    appState={
      Object {
        "isSettingsShowing": false,
        "isTokenDialogShowing": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "dialogs",
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        null,
      ],
      "className": "dialogs",
    },
    "ref": null,
    "rendered": Array [
      null,
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "dialogs",
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          null,
        ],
        "className": "dialogs",
      },
      "ref": null,
      "rendered": Array [
        null,
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
